lib/compiler.js

```diff
@@ -186,39 +186,82 @@ Compiler.prototype.visitTag = function (tag) {
  * Also, `virtual-dom` handles properties differently.
  */
 
+// attribute -> property transformations
+const ATTRS_TO_PROPS = {
+  'class': 'className',
+  'for': 'htmlFor',
+  'http-equiv': 'httpEquiv',
+}
+const LOWERCASE_ATTRS = new Set([
+  'acceptCharset',
+  'accessKey',
+  'allowFullScreen',
+  'allowTransparency',
+  'cellPadding',
+  'cellSpacing',
+  'colSpan',
+  'contentEditable',
+  'contextMenu',
+  'crossOrigin',
+  'dateTime',
+  'formAction',
+  'formEncType',
+  'formMethod',
+  'formNoValidate',
+  'formTarget',
+  'frameBorder',
+  'marginHeight',
+  'marginWidth',
+  'maxLength',
+  'mediaGroup',
+  'noValidate',
+  'readOnly',
+  'rowSpan',
+  'tabIndex',
+  'useMap',
+])
+for (const a of LOWERCASE_ATTRS) {
+  ATTRS_TO_PROPS[a.toLowerCase()] = a
+}
+
+function dashedToCaps(s) {
+  return s[1].toUpperCase()
```

Comment:
Nit but If you pass a second parameter it will be the first parenthesized match. Then you can just touppercase that and return it. It would make it a little more clear.